doc/apply_permutation.qbk

```diff
@@ -0,0 +1,85 @@
+[/ File apply_permutation.qbk]
+
+[section:apply_permutation apply_permutation]
+
+[/license
+Copyright (c) 2017 Alexander Zaitsev
+
+Distributed under the Boost Software License, Version 1.0.
+(See accompanying file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+]
+
+The header file 'apply_permutation.hpp' contains two algorithms, apply_permutation and apply_reverse_permutation. Also there are range-based versions. 
+The algorithms transform the item sequence according to index sequence order.
+
+The routine `apply_permutation` takes a item sequence and a order sequence. It reshuffles item sequence according to order sequence. Every value in order sequence means where the item comes from.
+The routine `apply_reverse_permutation` takes a item sequence and a order sequence. It will reshuffle item sequence according to order sequence. Every value in order sequence means where the item goes to.
```

Comment:
You might want to expand on the difference between `apply_permutation` and `apply_reverse_permutation`. Are they inverse operations?  (Running the example suggests not - but the explanation isn't clear to me)

Reply:
Are these unclear?
Every value in order sequence means where the item comes from.
Every value in order sequence means where the item goes to.

Reply:
I thought that it's very clear explanation: "the item comes from" vs "the item goes to".